src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -215,9 +215,9 @@ internal sealed class DebugInfo
         private class DebugInfoComparer : IComparer<DebugInfo>
         {
             //We allow comparison between int and DebugInfo here
-            int IComparer<DebugInfo>.Compare(DebugInfo d1, DebugInfo d2)
+            int IComparer<DebugInfo>.Compare([AllowNull] DebugInfo d1, [AllowNull] DebugInfo d2)
             {
-                if (d1.Index > d2.Index) return 1;
+                if (d1!.Index > d2!.Index) return 1; // TODO2
```

Comment:
:memo: `IComparer` declares that it can compare `null` inputs, but this implementation doesn't handle.

Reply:
It's only ever used with non-null values: https://source.dot.net/#System.Linq.Expressions/System/Linq/Expressions/Interpreter/LightCompiler.cs,233